### PR TITLE
sort the job run based on endtime and scheduled time

### PR DESCRIPTION
**Pull request by chlgit**
The job runs are sorted by end_time, and then scheduled time. It covers two edge cases: (1) rerun job, which has the same scheduled time with the first job run. (2) manually job run between two scheduled job runs. The details of these edge cases are in TRON-359.

### CODE REVIEW

File: tests/bin/check_tron_jobs_test.py [hunk 1]
```diff
@@ -439,6 +454,11 @@ def test_most_recent_job_with_smaller_run_id_failed(self):
                             '%Y-%m-%d %H:%M:%S',
                             time.localtime(time.time() - 1800),
                         ),
+                    'end_time':
+                        time.strftime(
+                            '%Y-%m-%d %H:%M:%S',
+                            time.localtime(time.time() - 1700),
```
Comment (qui) [inline tests/bin/check_tron_jobs_test.py]: Maybe this test should be called most_recent_end_time, since the run ID doesn't matter?

File: tests/bin/check_tron_jobs_test.py [hunk 2]
```diff
@@ -562,6 +669,7 @@ def test_job_running_but_action_unknown_already(self):
                             '%Y-%m-%d %H:%M:%S',
                             time.localtime(time.time() - 1800),
                         ),
+                    'end_time': None,
```
Comment (qui) [inline tests/bin/check_tron_jobs_test.py]: This shouldn't be None because it has succeeded.

File: bin/check_tron_jobs.py [hunk 3]
```diff
@@ -156,7 +156,7 @@ def pretty_print_actions(action_run):
 def get_relevant_run_and_state(job_content):
     job_runs = sorted(
         job_content.get('runs', []),
-        key=lambda k: k['run_time'],
+        key=lambda k: (k['end_time'] is None, k['end_time'], k['run_time']),
```
Comment (qui) [inline bin/check_tron_jobs.py]: I wrote out a few examples for myself, and it seems like this sort key translates into:
1) Currently running ones are always checked first (in case an action is failed/unknown)
2) If there are multiple running ones, then most recent run_time wins
3) If nothing is currently running, then most recent end_time wins

Is that correct? Can we add a comment about the intended behavior so we/others can remember in the future?

Comment (chlgit) [reply bin/check_tron_jobs.py]: Yes, it's right. I will make a comment in the code.

File: tests/bin/check_tron_jobs_test.py (hunk 2 above)
Comment (chlgit) [reply tests/bin/check_tron_jobs_test.py]: Oops, nice catch.

File: tests/bin/check_tron_jobs_test.py (hunk 1 above)
Comment (chlgit) [reply tests/bin/check_tron_jobs_test.py]: Sure, that makes sense.

**Comment by chlgit** [comment]
I reorganize these test cases by the monitoring states. All tests with the same state, such as STATE.STUCK, are grouped together.
